### ISSUE TITLE
Can't upload layer file that exported from attack-website

### ISSUE THREAD

**Issue by dickens88** [mitre-attack/attack-navigator#343]
Hi,
I deployed an attack-navigator with Docker in my local server, i had attack-website as well. I want to upload a layer file to navigator which was exported from attack)website, but there is an error when i try to upload.

![image](https://user-images.githubusercontent.com/8390209/130601734-ca728219-0a87-4401-88c5-fe4f7b64703b.png)

the layer file is downloaded from attack-web, ex, click Group> APT21 > download. I also try to upload the same file to navigator live version and it works fine.  is there any known issue about this?

![image](https://user-images.githubusercontent.com/8390209/130602093-0a28bec3-3b1d-41aa-9e0c-331d436f4a53.png)

**Comment by isaisabel** [comment]
Hi @dickens88,

I'm pretty sure the issue here is that your custom instance of ATT&CK Navigator has a custom value for the `versions` structure in the config file, and cannot match the `attack.versions` field in the layer you're uploading. 

If you're using a custom dataset (which I'm guessing you are) you should be able to simply remove the `versions.attack` field from the layer entirely and it _should_ then be able to upload (without the ATT&CK version specified it'll load the most "recent" version of the given domain). 

The following updates to `attack-navigator` and `attack-website` should help avoid this in the future:
- `attack-website`: Add a config option to build Navigator layers without the `versions.attack` field if using a custom dataset.
- `attack-navigator`: If when loading a layer it cannot match the specified `versions.attack` to any of the loaded versions, prompt the user to fallback to the current (most recent) ATT&CK version.

**Comment by isaisabel** [comment]
If you post one of the offending layer files and your ATT&CK Navigator config file I can verify if that's the actual cause, I'm pretty sure it is but there's a chance it's something else.

**Comment by dickens88** [comment]
Dear @isaisabel,

Just like you said, I'm using a custom dataset and the dataset is from Attack-Workbench. The configuration file looks like [this](https://github.com/center-for-threat-informed-defense/attack-workbench-frontend/blob/master/docs/integrations.md):
```
{
    "versions": [
        {
            "name": "ATT&CK Workbench Data", 
            "domains": [
                {   
                    "name": "Enterprise", 
                    "data": ["http://localhost:3000/api/stix-bundles/?domain=enterprise-attack"]
                },
                {   
                    "name": "Mobile", 
                    "data": ["http://localhost:3000/api/stix-bundles/?domain=mobile-attack"]
                },
                {
                    "name": "ICS",
                    "data": ["http://localhost:3000/api/stix-bundles/?domain=ics-attack"]
                }
            ]
        }
    ]
}
```
after i removed this block of code from the configuration file, then the error is gone.  I also tried to remove the `versions.attack` but it seems it doesn't work, can still meet the error.

the layer file I used can be download from offical ATT&CK matrix, the link is here: https://attack.mitre.org/groups/G0006/G0006-enterprise-layer.json

**Comment by isaisabel** [comment]
Yes that seems to be another bug. The code to get the "current version" (if a version is not specified in the layer file) for some reason seems to hardcode the version format. It also probably won't work properly with the release to ATT&CK v10 later this year since it will only match versions which have 1 digit (e.g ATT&CK v9 will match, ATT&CK v10 will not be matched).

https://github.com/mitre-attack/attack-navigator/blob/master/nav-app/src/app/data.service.ts#L290-L295

**Comment by clemiller** [comment]
The changes for [#370](https://github.com/mitre-attack/attack-navigator/issues/370) were included in the latest Navigator v4.6.0 release and appear to have addressed this issue. Let us know if this problem still persists for you. Closing.
